crates/esthri/src/ops/upload.rs

```diff
@@ -116,103 +119,14 @@ where
         file.as_ref().display()
     );
 
-    let compressed = false;
-    upload_file_helper(
-        s3,
-        bucket.as_ref(),
-        key.as_ref(),
-        file.as_ref(),
-        compressed,
-        Config::global().storage_class(),
-    )
-    .await
-}
-
-#[logfn(err = "ERROR")]
-pub async fn upload_with_storage_class<T>(
-    s3: &T,
-    bucket: impl AsRef<str>,
-    key: impl AsRef<str>,
-    file: impl AsRef<Path>,
-    storage_class: S3StorageClass,
-) -> Result<()>
-where
-    T: S3,
-{
-    info!(
-        "put: bucket={}, key={}, file={}",
-        bucket.as_ref(),
-        key.as_ref(),
-        file.as_ref().display()
-    );
-
-    let compressed = false;
     upload_file_helper(
         s3,
         bucket.as_ref(),
         key.as_ref(),
         file.as_ref(),
-        compressed,
-        storage_class,
-    )
-    .await
-}
-
-#[logfn(err = "ERROR")]
-pub async fn upload_compressed_with_storage_class<T>(
-    s3: &T,
-    bucket: impl AsRef<str>,
-    key: impl AsRef<str>,
-    file: impl AsRef<Path>,
-    storage_class: S3StorageClass,
-) -> Result<()>
-where
-    T: S3,
-{
-    info!(
-        "put: bucket={}, key={}, file={}",
-        bucket.as_ref(),
-        key.as_ref(),
-        file.as_ref().display()
-    );
-
-    let compressed = true;
-    upload_file_helper(
-        s3,
-        bucket.as_ref(),
-        key.as_ref(),
-        file.as_ref(),
-        compressed,
-        storage_class,
-    )
-    .await
-}
-
-#[logfn(err = "ERROR")]
-pub async fn upload_compressed<T>(
-    s3: &T,
-    bucket: impl AsRef<str>,
-    key: impl AsRef<str>,
-    file: impl AsRef<Path>,
-) -> Result<()>
-where
-    T: S3,
-{
-    info!(
-        "put(compressed): bucket={}, key={}, file={}",
-        bucket.as_ref(),
-        key.as_ref(),
-        file.as_ref().display()
-    );
-
-    let compressed = true;
-    upload_file_helper(
-        s3,
-        bucket.as_ref(),
-        key.as_ref(),
-        file.as_ref(),
-        compressed,
-        Config::global().storage_class(),
+        opts.transparent_compression,
+        opts.storage_class
```

Comment:
You can set default values in the `derive_builder` which might be an alternative to this.